crates/eframe/src/web/web_runner.rs

```diff
@@ -193,20 +191,47 @@ impl WebRunner {
     }
 
     pub(crate) fn request_animation_frame(&self) -> Result<(), wasm_bindgen::JsValue> {
+        if self.frame.borrow().is_some() {
+            // there is already an animation frame in flight
+            return Ok(());
+        }
+
         let window = web_sys::window().unwrap();
         let closure = Closure::once({
             let runner_ref = self.clone();
-            move || events::paint_and_schedule(&runner_ref)
+            move || {
+                // we can paint now, so clear the animation frame
+                // this drop the `closure` and allows another
+                // animation frame to be scheduled
+                let _ = runner_ref.frame.take();
```

Comment:
```suggestion
                runner_ref.frame = None;
```

Reply:
This doesn't work because it's an `Rc<RefCell<Option<...>>>`